lib/pipeline.js

```diff
@@ -10,11 +11,16 @@ class Pipeline extends Readable {
     this.basePath = basePath
     this.variables = variables
 
-    this.context = context
+    this.context = { ...context }
     this.context.basePath = this.basePath
     this.context.pipeline = this
     this.loaderRegistry = loaderRegistry
 
+    this.context.log = new Logger(this.node)
+    if (context.log) {
+      this.context.log.pipe(context.log)
```

Comment:
This pattern is also used in other places. What do you think about adding this feature already to the constructor of the `Logger`? So it would only require to call `new Logger(this.node, { master: this.context.log })`. If the master parameter is `null`/`undefined` it would be ignored.

Reply:
Of course. Makes sense :+1: